File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int calculate_letters(char *);
int calculate_words(char *);
int calculate_sentences(char *);
int nr_of_letters;
int nr_of_words;
int nr_of_sentences;


int main(void)
{
    char *text = get_string("Type some text: ");

    nr_of_letters = calculate_letters(text);
    nr_of_words = calculate_words(text);
    nr_of_sentences = calculate_sentences(text);

    // Average number of letters per 100 words
    float L = 100.00 * nr_of_letters /  nr_of_words;
    // Average number of sentences per 100 words
    float S = 100.00 * nr_of_sentences / nr_of_words;

    // Formula for Coleman-Liau index, which calculates
    // readability of given text
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int round_index = round(index);

    if (round_index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (round_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", round_index);
    }
}

// Function calculates number of letters in given text
int calculate_letters(char *text)
{
    nr_of_letters = 0;
    int str_length = strlen(text);
    for (int i = 0; i < str_length; i++)
    {
        if (isalpha(text[i]))
        {
            nr_of_letters++;
        }
    }
    return nr_of_letters;
}

// Function calculates number of words in given text
int calculate_words(char *text)
{
    int empty_spaces = 0;
    int str_length = strlen(text);
    for (int i = 0; i < str_length; i++)
    {
        if (text[i] == ' ')
        {
            empty_spaces += 1;
            nr_of_words = empty_spaces + 1;
        }
    }
    return nr_of_words;
}

// Function calculates number of sentences in given text
int calculate_sentences(char *text)
{
    int str_length = strlen(text);
    int full_stops = 0;
    nr_of_sentences = 0;
    for (int i = 0; i < str_length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            nr_of_sentences++;
        }
    }
    return nr_of_sentences;
}

File: pset6/readability/readability.py
text = input("Please enter some text: ")


def count_letters(text):
    """Function calculates number of letters in given text"""
    letter_count = 0
    for i in text:
        if i.isalnum():
            letter_count += 1
    return letter_count


def count_words(text):
    """Function calculates number of words in given text"""
    empty_spaces = 0
    word_count = 0
    for i in range(0, len(text)):
        if text[i] == ' ':
            empty_spaces += 1
            word_count = empty_spaces + 1
    return word_count


def count_sentences(text):
    """Function calculates number of sentences in given text"""
    sentence_count = 0
    for i in range(0, len(text)):
        if text[i] == '.' or text[i] == '?' or text[i] == '!':
            sentence_count += 1
    return sentence_count


def coleman_liau_index(letter_count, word_count, sentence_count):
    """Formula for Coleman-Liau index, which calculates
       readability of given text"""
    # Average number of letters per 100 words
    l = 100.00 * letter_count / word_count
    # Average number of sentences per 100 words
    s = 100.00 * sentence_count / word_count
    index = round(0.0588 * l - 0.296 * s - 15.8)
    return index


def print_index(index):
    if index > 16:
        print("Grade 16+", end="\n")
    elif index < 1:
        print("Before Grade 1", end="\n")
    else:
        print("Grade: ", index, end="\n")


def main():
    letter_count = count_letters(text)
    word_count = count_words(text)
    sentence_count = count_sentences(text)
    coleman_liau_index(letter_count, word_count, sentence_count)
    index = coleman_liau_index(letter_count, word_count, sentence_count)
    print_index(index)


main()

